sup/constants.py

```diff
@@ -1,11 +1,20 @@
 import pathlib
-CONFIG_PATH = pathlib.Path("/opt/MagAOX/config")
-REPLICATED_CAMERAS = [
-    'camsci1',
-    'camsci2',
-    'camwfs',
-    # 'camlowfs',
-    'camtip',
-    # 'camacq',
-]
-TMPFILE_ROOT = pathlib.Path("/dev/shm")
\ No newline at end of file
+
+import json
+
+with open('../constants.json', 'r') as file:
```

Comment:
As written, this isn't working... `../` is relative to the sup process current working directory, which we don't expect to be `/opt/MagAOX/source/sup/sup`. Also, we always install `sup` in editable mode, but I don't think we should depend on that being true here.

Reply:
Yes, that's a good point. I can see three solutions:

1. Assert separation of concerns and go back to decoupling front and back end constants. Put any constants used by both in the config file. Default values for config file options are set in the backend constants file and the frontend assumes it will always receive a value.
2. Move the consolidated json constants file in the backend, read it there and pass all variables required by the frontend through the fetch call.
3. Move all constants in the config file. The backend reads it and passes all variables required by the frontend through the fetch call. This would need to enforce a minimum level of config present in the config file, since there wouldn't be any default values.

Do you have a preferred solution or a convention that you're already using? 
I think there's no ideal solution and it depends on what we're trying to optimize. I'm leaning towards the first approach, going back to two constants files.

Reply:
I agree there's no perfect solution, but I favor option 1. If that requires too many onerous gymnastics to get the frontend configured, we can also just live with redundancy for now and use elbow grease to keep two constants files in sync. We don't mix and match frontend and backend versions, so they should evolve in sync.

Reply:
Cool, I've made these changes.

I considered dynamically building the tab bar import paths from the layout variable, which is why I converted the config values and folder names to lowercase ('magaox' & 'scoob'). However that would require including Webpack as a dependency and we wouldn't fully eliminate manual input anyway, since the routes would still need to be mapped manually to the corresponding layout (what is now in map.js). In the end it doesn't seem worth the effort at this stage, but it's still an option in the future. But I kept the lowercase convention to have it all self-consistent.